### ISSUE TITLE
[TreeListBox] Fix144# has introduced a new bug in HierarchySourceChanged event in case of Single SelectionMode

### ISSUE THREAD

**Issue by PIUM-arc** [PropertyTools/PropertyTools#301]
The commit "Fix #144 + various fixes" done in 3.0 has introduced a new bug in the HierarchySourceChanged method. 
The code changed from 

```
if (oldTreeSource != null)
{
	foreach (var item in oldTreeSource)
	{
		var container = this.GetContainerFromItem(item);
		if (container == null)
		{
			continue;
		}

		if (container.IsSelected)
		{
			this.SelectedItems.Remove(item);
		}
	}
}
```

to 
```
if (oldTreeSource != null)
{
	this.SelectedItems.Clear();
}
```

The problem is, SelectedItems throw an InvalidOperationException in case of SelectionMode set to Single 
see : https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.listbox.selecteditems?view=windowsdesktop-6.0#system-windows-controls-listbox-selecteditems.

**Comment by PIUM-arc** [comment]
Hello, 
Nobody ?

**Comment by PIUM-arc** [comment]
Hello,
I just did a pull request regarding this issue.

**Comment by PIUM-arc** [comment]
Hello,
Sorry to insist but I really really need this fix to be integrated.
https://github.com/PropertyTools/PropertyTools/pull/304

**Comment by PIUM-arc** [comment]
Solved and integrated in commit : f21d095fe3dd5a92aa58c12930dfbf4bd998ebeb
